libraries/MQTT/MQTT.cpp

```
@@ -48,17 +48,88 @@ void MQTT::callback(char* topic, byte* payload, unsigned int length) {
 /*  This should be called in loop function in main device file  */
 void MQTT::loop(){
     if(WiFi.isConnected())
-        if (!client.connected())
+        if (!client.connected()) {
+            /* Load a certificate before connection, the ESP clock is set */
+            if(enabled_tls)
+                load_certificate();
             reconnect();
-    
+        }
+
     client.loop();
 }
 
 /*  In constructot a library configures mqtt and setup callback function  */
 MQTT::MQTT(){
     client.setClient(wifiClient);
     client.setServer(broker, port);
-
+    enabled_tls = true;
     std::function<void(char*, uint8_t*, unsigned int)> WhereCatsDoMEWMEW = std::bind(&MQTT::callback, this,std::placeholders::_1,std::placeholders::_2,std::placeholders::_3);
     client.setCallback(WhereCatsDoMEWMEW);
 }
+
+void MQTT::load_certificate() {
```

This method should return an error code in case something goes south.

---

done